File: python/0118.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        ans=[[0]*numRows for i in range(numRows)]
        for i in range(numRows):
            ans[i][0]=1
        for i in range(1,numRows):
            for j in range(1,i+1):
                ans[i][j]=ans[i-1][j]+ans[i-1][j-1]
        a=[ans[i][:(i+1)]for i in range(numRows)]
        return a

File: rust/src/bin/0118.rs
struct Solution;

impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        if num_rows == 0 {
            return vec!();
        }
        let mut a = vec!();
        a.push(vec!(1));
        for i in 1..num_rows {
            let mut now = vec!(1);
            let last = a.last().unwrap();
            for j in 1..=i as usize {
                now.push(last[j - 1] + if j >= last.len() { 0 } else { last[j] });
            }
            a.push(now);
        }
        a
    }
}

fn main() {
    println!("{:?}", Solution::generate(5));
}